### PR TITLE
TabView: API changes to match spec

### PR DESCRIPTION
**Pull request by teaP**
This change is mostly changing names to match the spec, but there are some functionality changes as well. With this change, TabView is moved out of preview. The list of edits is:

TabView
	AddButtonCommand -> AddTabButtonCommand
	AddButtonCommandParameter -> AddTabButtonCommandParameter
	IsAddButtonVisible -> IsAddTabButtonVisible
	CanDragDropTabs -> split into CanDragTabs, CanReorderTabs, AllowDropTabs
	ItemsSource -> TabItemsSource
	Items -> TabItems
	ItemTemplate -> TabItemTemplate
	ItemTemplateSelector -> TabItemTemplateSelector
	LeftCustomContent -> TabStripHeader
	LeftCustomContentTemplate -> TabStripHeaderTemplate
	RightCustomContent -> TabStripFooter
	RightCustomContentTemplate -> TabStripFooterTemplate
	AddButtonClick -> AddTabButtonClick
	TabClosing -> TabCloseRequested
	TabDraggedOut -> TabDroppedOutside
	TabStripDragItemsStarting -> TabDragStarting
	TabStripDragItemsCompleted -> TabDragCompleted
	Change SelectionChanged sender to be TabView
	TabClosing -> TabCloseRequested
TabViewItem
	IsCloseable -> IsClosable
	TabClosing -> CloseRequested
Event Args
	TabClosingEventArgs -> TabCloseRequestedEventArgs
		Remove Cancel. Add TabViewItem parameter.
	Create event args for DragStarting/DragCompleted events

I fixed up all test affected by this and also added a new test that proves you can drag/drop between TabViews.

### CODE REVIEW

File: dev/TabView/TabView.idl [hunk 1]
```diff
@@ -1,67 +1,89 @@
 ﻿namespace MU_XC_NAMESPACE
 {
 
-[WUXC_VERSION_PREVIEW]
+[WUXC_VERSION_MUXONLY]
 [webhosthidden]
 enum TabViewWidthMode
 {
     SizeToContent = 0,
```
Comment (MikeHillberg) [inline dev/TabView/TabView.idl]: The zero value should be Equal now, since it's the default value.

Comment (teaP) [reply dev/TabView/TabView.idl]: Fixed!

File: dev/TabView/TabView.cpp [hunk 2]
```diff
@@ -325,13 +326,46 @@ void TabView::OnListViewSelectionChanged(const winrt::IInspectable& sender, cons
...
 
-    m_selectionChangedEventSource(sender, args);
+    m_selectionChangedEventSource(*this, args);
 }
 
+winrt::TabViewItem TabView::FindTabViewItemFromDragItem(const winrt::IInspectable& item)
+{
+    auto tab = ContainerFromItem(item).try_as<winrt::TabViewItem>();
+
+    if (!tab)
```
Comment (kmahone) [inline dev/TabView/TabView.cpp]: What are the cases where ContainerFromItem will fail to give the required container?

Comment (teaP) [reply dev/TabView/TabView.cpp]: Unfortunately, this happens during drag and drop. There's a period of time after the drop where ContainerFromItem will NOT return the item even if it is a TabViewItem. Once we switch to Repeater hopefully we can get rid of a lot of this ridiculousness.

File: dev/TabView/TabView.h [hunk 3]
```diff
@@ -8,36 +8,70 @@
...
+    winrt::IInspectable m_item{};
+    tracker_ref<winrt::TabViewItem> m_tab{ this };
+};
 
+class TabViewTabDragStartingEventArgs :
+    public ReferenceTracker<TabViewTabDragStartingEventArgs, winrt::implementation::TabViewTabDragStartingEventArgsT, winrt::composing, winrt::composable>
+{
+public:
+    TabViewTabDragStartingEventArgs(winrt::DragItemsStartingEventArgs const& args, winrt::IInspectable const& item, winrt::TabViewItem tab) { m_args = args; m_item = item; m_tab.set(tab); }
+
```
Comment (jevansaks) [inline dev/TabView/TabView.h]: Wrap lines, and [prefer initialization to assignment](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c49-prefer-initialization-to-assignment-in-constructors):

```c++
: m_args(args), m_item(item), m_tab(tab)
```

Comment (teaP) [reply dev/TabView/TabView.h]: OK, done.

File: dev/TabView/TabView.h [hunk 4]
```diff
@@ -18,7 +18,10 @@ class TabViewTabCloseRequestedEventArgs :
     public ReferenceTracker<TabViewTabCloseRequestedEventArgs, winrt::implementation::TabViewTabCloseRequestedEventArgsT, winrt::composing, winrt::composable>
 {
 public:
-    TabViewTabCloseRequestedEventArgs(winrt::IInspectable const& item, winrt::TabViewItem tab) { m_item = item; m_tab.set(tab); }
+    TabViewTabCloseRequestedEventArgs(winrt::IInspectable const& item, winrt::TabViewItem tab) : m_item(item)
+    {
+        m_tab.set(tab);
```
Comment (jevansaks) [inline dev/TabView/TabView.h]: The event args are short-lived and don't risk creating a cycle so they don't need to be ReferenceTracker or tracker_ref. Can you downgrade them? Just do:
```c++
TabViewTabCloseRequestedEventArgs : public winrt::implementation::TabViewTabCloseRequestedEventArgsT<TabViewTabCloseRequestedEventArgs>
```

That should simplify these classes a bit.

Comment (teaP) [reply dev/TabView/TabView.h]: Sure, can do.

**Pull request merged**